hips/hip-0019.md

```diff
@@ -0,0 +1,63 @@
+
+#### Example
+```
+# .helmlintconfig.yaml file example
+lintIgnore:
+  - pathIgnore: "migrations/templates/job.yaml"                                       # Ignores all errors from this file
+  - errorIgnore: "chart metadata is missing these dependencies**"                     # Ignores all icon errors
+  - pathIgnore: "gitlab/templates/shared-secrets/self-signed-cert-job.yml"
+    errorIgnore: "<include 'fullname' .>"                                             # Ignores this specific error but not other errors in the file
+```
+
+### Command Behavior
+When `helm lint` is executed, it checks for the presence of a `.helmlintconfig.yaml` file in the chart directory or at a location specified by the `--lint-config-file` flag. The command then parses this file to exclude any files or directories matching the patterns specified, including specific error lines.
+
+## Backwards Compatibility
+This proposal introduces no breaking changes. Charts without a `.helmlintconfig.yaml` file or those not using the `--lint-config-file` flag will function as before.
+
+## Security Implications
+No significant security implications are expected as the `.helmlintconfig.yaml` file is processed locally by `helm lint`, without modifying chart content or involving network activity.
+
+## How to Teach This
+Documentation for the `.helmlintconfig.yaml` file and the `--lint-config-file` flag will be integrated into the official Helm documentation under the `helm lint` section. Detailed examples and common patterns will be provided to assist new users.
+
+## Reference Implementation
+PR #[13257](https://github.com/helm/helm/pull/13257) created in the Helm/helm repository in gitlab 
+
+## Open Issues
+- Developing an Effective and User-Friendly Pattern Syntax: We are considering the design of a flexible and intuitive pattern syntax for the `.helmlintconfig.yaml` file. This syntax would ideally support rule-specific configurations and exclusions, allowing users to finely control lint behaviors on a per-rule basis.
+- In-File Annotations for Excluding Chart Sections: Exploring the possibility of in-file annotations that would let developers temporarily exclude specific sections of a chart from linting. This feature would provide granular control directly within Helm chart files.
+- Rule Naming and Configurations: Currently, Helm does not support named lint rules. Introducing named rules is a foundational step that would enable us to implement more sophisticated configuration features, such as rule-specific exclusions and settings. This would also allow users to extend or override a "default" configuration set, similar to practices seen in tools like [yamllint](https://yamllint.readthedocs.io/en/stable/configuration.html#extending-the-default-configuration). Naming rules would facilitate clearer and more maintainable configurations, laying the groundwork for future enhancements.
```

Comment:
Looks like bullet points 1 and 3 here might be overlapping, maybe we could consolidate?